### PR TITLE
kh360-react-eval-screen-select-limited-10-items

### PR DESCRIPTION
**Pull request by chadiegilnerubia**
Ticket ID:
[Evaluation Result: Evaluation Administration Filter - All Closed and Published should be display in the dropdown selection#504](https://github.com/nerubia/kh360-react/issues/504)

### CODE REVIEW

**Review by gborgonia-ir — APPROVED** [review]

File: src/components/ui/select/custom-select-infinite.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    )
+  })
+
+export const CustomSelectInfinite = async (
+  search: string,
+  prevOptions: OptionsOrGroups<Option, GroupBase<Option>>,
+  evaluationAdministrationFilters: Option[],
+  loading: boolean,
+  totalEvalItems: number,
+  hastNextPage: boolean
```
Comment (emmaacha) [inline src/components/ui/select/custom-select-infinite.tsx:20]: Typo for `hasNextPage`

File: src/components/ui/select/custom-select-infinite.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  let filteredOptions: Option[]
+  if (search.length === 0) {
+    filteredOptions = options
+  } else {
+    const searchLower = search.toLowerCase()
+    filteredOptions = options.filter(({ label }) => label.toLowerCase().includes(searchLower))
+  }
+
+  const hasMore = totalEvalItems > 10 || hastNextPage
```
Comment (emmaacha) [inline src/components/ui/select/custom-select-infinite.tsx:37]: Typo for `hasNextPage`

File: src/features/evaluation-results/evaluation-results-list-filter.tsx [hunk 3]
```diff
@@ -68,29 +70,59 @@ export const EvaluationResultsListFilter = () => {
   const [banding, setBanding] = useState<string>(searchParams.get("banding") ?? "all")
   const [sortBy, setSortBy] = useState<string>(searchParams.get("sort_by") ?? "evaluee")
   const isMobile = useMobileView()
+  const [page, setPage] = useState<string>("1")
+  const [hastNextPage, setHasNextPage] = useState<boolean>(false)
```
Comment (emmaacha) [inline src/features/evaluation-results/evaluation-results-list-filter.tsx:74]: Typo for `hasNextPage`

File: src/features/evaluation-results/evaluation-results-list-filter.tsx [hunk 4]
```diff
@@ -68,29 +70,59 @@ export const EvaluationResultsListFilter = () => {
...
       label: evalAdmin.name ?? "",
       value: evalAdmin.id.toString(),
     }))
-    filterOptions.unshift({
-      label: "All",
-      value: "all",
-    })
-    setEvaluationAdministrationFilters(filterOptions)
+
+    if (hastNextPage) {
```
Comment (emmaacha) [inline src/features/evaluation-results/evaluation-results-list-filter.tsx:116]: Typo for `hasNextPage`

File: src/features/evaluation-results/evaluation-results-list-filter.tsx [hunk 5]
```diff
@@ -142,18 +173,32 @@ export const EvaluationResultsListFilter = () => {
...
                 (option) => option.value === evaluationAdministrationId
               )}
+              loadOptions={async (search, prevOptions) =>
+                await CustomSelectInfinite(
+                  search,
+                  prevOptions,
+                  evaluationAdministrationFilters,
+                  loading,
+                  totalEvalItems,
+                  hastNextPage
```
Comment (emmaacha) [inline src/features/evaluation-results/evaluation-results-list-filter.tsx:196]: Typo for `hasNextPage`

**Review by emmaacha — CHANGES_REQUESTED** [review]

File: src/components/ui/select/custom-select-infinite.tsx (hunk 1 above)
Comment (chadiegilnerubia) [reply src/components/ui/select/custom-select-infinite.tsx]: Donee thanks :3

**Review by emmaacha — APPROVED** [review]

**Pull request merged**